src/Scripting/Core/Hosting/CommandLine/CommandLineRunner.cs

```diff
@@ -109,14 +109,17 @@ private int RunInteractiveCore(ErrorLogger errorLogger)
             }
 
             var scriptPathOpt = sourceFiles.IsEmpty ? null : sourceFiles[0].Path;
-            var scriptOptions = GetScriptOptions(_compiler.Arguments, scriptPathOpt, _compiler.MessageProvider, diagnosticsInfos);
 
             var errors = _compiler.Arguments.Errors.Concat(diagnosticsInfos.Select(Diagnostic.Create));
             if (_compiler.ReportErrors(errors, _console.Error, errorLogger))
             {
                 return CommonCompiler.Failed;
             }
 
+            // only emit symbols for non-interactive mode, and when encoding is known
+            var emitDebugInformation = !_compiler.Arguments.InteractiveMode && code.Encoding != null;
```

Comment:
Why not emit debug info in interactive mode? The line info would be useful when you #load a script, e.g.

---
In reply to: [103508026](https://github.com/dotnet/roslyn/pull/17379#discussion_r103508026) [](ancestors = 103508026)